File: 78-subsets/subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        
        List<List<Integer>> res = new ArrayList<>();
        res.add(new ArrayList<>());

        for (int num : nums) {
            int currentSize = res.size();

            for (int i = 0; i < currentSize; i++) {

                List<Integer> subSet = new ArrayList<>(res.get(i));
                subSet.add(num);
                res.add(subSet);
            }
        }
        return res;
    }
}

File: 78-subsets/subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
        let res = [[]];


    for(let i=0;i<nums.length;i++) {
        let curr = nums[i];
        resLen = res.length;

        for(let j=0;j<resLen;j++) {

            let subSet = [...res[j],curr];
            res.push(subSet);
        }
    } return res;
    
};